packages/pluggableWidgets/rich-text-web/src/components/RichText.tsx

```diff
@@ -0,0 +1,112 @@
+import { createElement, ReactElement, useState, useEffect, useMemo } from "react";
+import { CKEditorHookProps, CKEditorType, CKEditorConfig, CKEditorEventAction } from "ckeditor4-react";
+import { getDimensions, Dimensions } from "@mendix/piw-utils-internal";
+import { defineEnterMode, addPlugin, PluginName } from "../utils/ckeditorConfigs";
+import sanitizeHtml from "sanitize-html";
+import classNames from "classnames";
+import { ReadOnlyStyleEnum, EnterModeEnum, ShiftEnterModeEnum, AdvancedConfigType } from "../../typings/RichTextProps";
+import { MainEditor } from "./MainEditor";
+
+export interface RichTextProps {
+    name: string;
+    readOnly: boolean;
+    spellChecker: boolean;
+    sanitizeContent?: boolean;
+    value: string | undefined;
+    advancedConfig: AdvancedConfigType[] | null;
+    plugins?: string[];
+    readOnlyStyle: ReadOnlyStyleEnum;
+    toolbar: CKEditorConfig;
+    enterMode?: EnterModeEnum;
+    shiftEnterMode?: ShiftEnterModeEnum;
+    editorType: CKEditorType;
+    dimensions?: Dimensions;
+    advancedContentFilter?: { allowedContent: string; disallowedContent: string } | null;
+    onValueChange?: (value: string) => void;
+    onKeyChange?: () => void;
+    onKeyPress?: () => void;
+}
+
+export const RichTextEditor = (props: RichTextProps): ReactElement => {
+    const {
+        editorType,
+        plugins,
+        enterMode,
+        shiftEnterMode,
+        value,
+        readOnly,
+        readOnlyStyle,
+        advancedContentFilter
+    } = props;
+    const [element, setElement] = useState<HTMLElement | null>(null);
+    const { width, height } = props.dimensions
+        ? getDimensions({ ...props.dimensions })
+        : {
+              width: "100%",
+              height: "100%"
+          };
+    const [ckeditorConfig, setCkeditorConfig] = useState<CKEditorHookProps<"change">>({
+        element,
+        editorUrl: "/widgets/ckeditor/ckeditor.js",
```

Comment:
Maybe we can do better and rely on `${window.mx.remoteUrl}/widgets/...` to assure the URL is correct.

Reply:
how should I deal with typings error here? is there any example?

Reply:
You should implement the global for window with typings. Same as https://github.com/mendix/widgets-resources/blob/3dea5d093148b8e7fc1b26ed364b8a42036e013d/packages/pluggableWidgets/datagrid-date-filter-web/typings/global.d.ts#L26